src/test/run-make-fulldeps/coverage/issue-84561.rs

```diff
@@ -0,0 +1,166 @@
+// This demonstrated Issue #84561: function-like macros produce unintuitive coverage results.
```

Comment:
Can you add a test for nested macro invocations (both invoking a macro inside a macro invocation, and invoking another macro inside `macro_rules!`)? I don't care too much what the output looks like, as long as it's reasonable and doesn't cause an ICE :)

Reply:
I believe I have tests for the first case (see lines 104 and 110).

I'll add a test for the second case.

Reply:
Done